service/version.go

```diff
@@ -153,24 +153,30 @@ func (uis *UIServer) versionPage(w http.ResponseWriter, r *http.Request) {
 					DisplayName: t.DisplayName,
 				}}
 
-			// TODO: this loop would probably work better
-			// as an aggregation.
-			if t.Status == evergreen.TaskStarted {
-				var taskFromDb *task.Task
-				taskFromDb, err = task.FindOne(task.ById(t.Id))
-				if err != nil {
-					uis.LoggedError(w, r, http.StatusInternalServerError, err)
-				} else if taskFromDb != nil {
-					uiT.ExpectedDuration = taskFromDb.ExpectedDuration
-				}
-
-				grip.ErrorWhen(taskFromDb == nil, message.Fields{
+			var taskFromDb *task.Task
+			taskFromDb, err = task.FindOne(task.ById(t.Id).WithFields(task.FinishTimeKey, task.ExpectedDurationKey))
+			if err != nil {
+				uis.LoggedError(w, r, http.StatusInternalServerError, err)
+				return
+			}
+			if taskFromDb == nil {
+				grip.Error(message.Fields{
 					"task_id": t.Id,
 					"version": projCtx.Version.Id,
 					"request": gimlet.GetRequestID(ctx),
 					"message": "version references task that does not exist",
```

Comment:
not your code, but should read "_build_ references..."

Reply:
done.